packages/core/src/components/sections/ProductShelf/ProductShelf.tsx

```diff
@@ -1,22 +1,51 @@
+import { memo } from 'react'
 import { useInView } from 'react-intersection-observer'
 import Section from '../Section'
 
+import ProductShelf from 'src/components/ui/ProductShelf'
+import { useProductsQuery } from 'src/sdk/product/useProductsQuery'
 import styles from './section.module.scss'
-import ProductShelf, { ProductShelfProps } from 'src/components/ui/ProductShelf'
 
-function ProductShelfSection({
-  ...otherProps
-}: Omit<ProductShelfProps, 'inView'>) {
+type Sort =
+  | 'discount_desc'
+  | 'name_asc'
+  | 'name_desc'
+  | 'orders_desc'
+  | 'price_asc'
+  | 'price_desc'
+  | 'release_desc'
+  | 'score_desc'
+
+export type ProductShelfProps = {
+  title: string
+  first?: number
+  after?: string
+  sort?: Sort
+  term?: string
+  selectedFacets?: {
+    key: string
+    value: string
+  }[]
+  productCardConfiguration?: {
+    showDiscountBadge?: boolean
+    bordered?: boolean
+  }
+}
+
+function ProductShelfSection(props: ProductShelfProps) {
   const { ref, inView } = useInView()
+  const { first, after, sort, term, selectedFacets, ...otherProps } = props
+  const data = useProductsQuery({ first, after, sort, term, selectedFacets })
+  const products = data?.search?.products
 
   return (
     <Section
       className={`${styles.section} section-product-shelf layout__section`}
       ref={ref}
     >
-      <ProductShelf inView={inView} {...otherProps} />
+      <ProductShelf inView={inView} products={products} {...otherProps} />
     </Section>
   )
 }
 
-export default ProductShelfSection
+export default memo(ProductShelfSection)
```

Comment:
I'd remove this memo here and add one in RenderSections

Reply:
Nice! after adding the memo in the `RenderSections` I also test the rendering process from this section and didn't see differences, so the memo was removed [here](https://github.com/vtex/faststore/pull/1967/commits/f99e08d9934737022cb2e8317e6af709b2a87d7a).